lib/src/main/scala/spinal/lib/com/pmod/PMOD.scala

```diff
@@ -0,0 +1,23 @@
+package spinal.lib.com.pmod
+
+import spinal.core._
+
+abstract class PMODBundle extends Bundle {
+  val pin1 = Bits(1 bit)
+  val pin2 = Bits(1 bit)
+  val pin3 = Bits(1 bit)
+  val pin4 = Bits(1 bit)
+  val pin7 = Bits(1 bit)
+  val pin8 = Bits(1 bit)
+  val pin9 = Bits(1 bit)
+  val pin10 = Bits(1 bit)
+
```

Comment:
What about instead having Bits(8 bits) ?

Reply:
Do we can set direction for each bit  in Bits? We don't know which pins gonna be input and which ping gonna be output here.

Reply:
Hoooooooo
That's right, that's a issue :/ Only solution then is to use tristate pins. or as you did having separated signal for each pin.
So i would say let's go you way.